### PR TITLE
#3585 Show whitelisting stats during presale

### PR DESCRIPTION
**Pull request by desfero**

### CODE REVIEW

File: packages/web/app/components/issuer-dashboard/bookBuildingWidget/BookBuildingWidget.tsx [hunk 1]
```diff
@@ -89,7 +96,7 @@ const BookBuildingStats = ({ bookBuildingStats, maxPledges, downloadCSV }: IBook
       <span className={styles.label}>
         <FormattedMessage id="shared-component.eto-overview.investors-backed" />
       </span>
-      <span data-test-id="eto-bookbuilding-investors-backed">
+      <span data-test-id="bookbuilding-widget.stats.number-of-pledges">
```
Comment (drakkein) [inline packages/web/app/components/issuer-dashboard/bookBuildingWidget/BookBuildingWidget.tsx]: Can we hide whole span if maxPledges === null, instead just hiding message?

File: packages/web/app/components/issuer-dashboard/bookBuildingWidget/BookBuildingWidget.tsx [hunk 2]
```diff
@@ -103,108 +110,112 @@ const BookBuildingStats = ({ bookBuildingStats, maxPledges, downloadCSV }: IBook
...
-    return (
-      <BookBuildingWidgetLayout
-        headerText={<FormattedMessage id="settings.book-building-widget.book-building-enabled" />}
-        text={
-          <FormattedHTMLMessage
-            tagName="span"
-            id="settings.book-building-widget.book-building-enabled-text"
-            values={{ minOffsetPeriod: minOffsetPeriod / DAY }}
+  switch (whitelistingState) {
+    case EWhitelistingState.ACTIVE:
```
Comment (drakkein) [inline packages/web/app/components/issuer-dashboard/bookBuildingWidget/BookBuildingWidget.tsx]: I would extract every particular case as separate component for better readability, but it's not urgent

File: packages/web/app/components/issuer-dashboard/bookBuildingWidget/BookBuildingWidget.tsx [hunk 3]
```diff
@@ -270,4 +278,29 @@ export const BookBuildingWidget = compose<React.FunctionComponent<IExternalProps
     },
   }),
   branch<IStateProps>(props => !props.bookBuildingStats, renderComponent(WidgetLoading)),
+  branch<IStateProps>(
+    // show widget when bookbuilding can be enabled or when eto is in presale state (to still access stats)
+    props => !props.canEnableBookbuilding && props.onChainState !== EETOStateOnChain.Whitelist,
```
Comment (drakkein) [inline packages/web/app/components/issuer-dashboard/bookBuildingWidget/BookBuildingWidget.tsx]: would be nice to have it moved as utility function like `canEnableBookbuilding(canEnable, onChainState)`, but also not urgent

File: packages/web/app/test-e2e/eto/Campaigning.spec.e2e.ts [hunk 4]
```diff
@@ -58,57 +88,116 @@ describe("Eto campaigning state", () => {
...
-            cy.get<number>("@remainingSlots").then(remainingSlots => {
-              // Remove two from remaining slots as it's second pledge
-              cy.get(tid("eto-bookbuilding-remaining-slots")).should("contain", remainingSlots - 2);
-            });
-          });
+        });
+
+        cy.visit(etoPublicViewByIdLinkLegacy(ETO_ID));
+
+        cy.wait(5000); //let the store get the ETO data, otherwise this place is flaky
```
Comment (drakkein) [inline packages/web/app/test-e2e/eto/Campaigning.spec.e2e.ts]: oh no 😱 
but ok, for now

File: packages/web/app/components/issuer-dashboard/bookBuildingWidget/BookBuildingWidget.tsx (hunk 1 above)
Comment (desfero) [reply packages/web/app/components/issuer-dashboard/bookBuildingWidget/BookBuildingWidget.tsx]: fixed

File: packages/web/app/components/issuer-dashboard/bookBuildingWidget/BookBuildingWidget.tsx (hunk 2 above)
Comment (desfero) [reply packages/web/app/components/issuer-dashboard/bookBuildingWidget/BookBuildingWidget.tsx]: Done

File: packages/web/app/components/issuer-dashboard/bookBuildingWidget/BookBuildingWidget.tsx (hunk 3 above)
Comment (desfero) [reply packages/web/app/components/issuer-dashboard/bookBuildingWidget/BookBuildingWidget.tsx]: the code was already moved

File: packages/web/app/test-e2e/eto/Campaigning.spec.e2e.ts (hunk 4 above)
Comment (desfero) [reply packages/web/app/test-e2e/eto/Campaigning.spec.e2e.ts]: we are gonna refactor book building flow anyway to use modal so, for now, let's leave `wait` as we don't have any `loading indicator` to assert

**Pull request merged**